hbase-common/src/main/java/org/apache/hadoop/hbase/io/encoding/RowIndexEncoderV1.java

```diff
@@ -45,6 +45,8 @@ public int write(Cell cell) throws IOException {
         startOffset = out.size();
       }
       rowsOffsetBAOS.writeInt(out.size() - startOffset);
+      // added for the int written in the previous line
+      context.getEncodingState().encodedDataSizeWritten += 4;
```

Comment:
You can still use the postEncodeCell method here na?

Reply:
Done